File: 3754-ConcatenateNonZeroDigitsAndMultiplyBySumI/3754-ConcatenateNonZeroDigitsAndMultiplyBySumI.java
// Last updated: 9/14/2026, 11:54:07 PM
class Solution {
    public long sumAndMultiply(int n) {
        long x = 0,sum = 0,len = 1;
        while(n != 0){
            x = (n % 10) * len + x;
            if(n % 10 != 0) len = len * 10;
            sum += n % 10;
            n = n / 10;
        }
        return sum * x;
    }
}

File: 3754-ConcatenateNonZeroDigitsAndMultiplyBySumI/3754-ConcatenateNonZeroDigitsAndMultiplyBySumI.py
# Last updated: 9/14/2026, 11:54:12 PM
class Solution(object):
    def sumAndMultiply(self, n):
        x_str = ""

        for ch in str(n):
            if ch != '0':
                x_str += ch

        if x_str == "":
            return 0

        x = int(x_str)
        digit_sum = sum(int(c) for c in x_str)

        return x * digit_sum
